### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by psouquet**
create a functional image with version selection as argument

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+# Copy the script (first)
+COPY docker-wrapper.sh /opt/petals-cockpit/docker-wrapper.sh
+
+# FIXME: unable to get the latest tag of gitlab without specific tag :
+# see https://gitlab.com/gitlab-org/gitlab-ce/issues/22273
+# and https://gitlab.com/gitlab-org/gitlab-ce/issues/26254
+# For now latest tag must be updated each release of cockpit.
+# Important: temporary system packages are deleted at the end.
+RUN apt-get update -y && \
```
Comment (vincent-zurczak) [inline Dockerfile:26]: 3 RUN means 3 layers.
Not great.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  apt-get install curl -y && \
+  apt-get upgrade -y 
+RUN if [ "${COCKPIT_VERSION}" = "LATEST" ]; then DOWNLOAD_URL="${DOWNLOAD_URL}v0.21.0/download?job=release-product"; \
```
Comment (vincent-zurczak) [inline Dockerfile]: Try to replace it by a one-liner.
Example: `([ "${COCKPIT_VERSION}" == "LATEST" ] && DOWNLOAD_URL="") || ([ "${COCKPIT_VERSION}" == "SNAPSHOT" ] && DOWNLOAD_URL="")`

And then merge all the RUN instructions into a single one.

File: docker-wrapper.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+# Place holder script to configure cockpit environment
+#
+# To prevent this thing to happen on every start, we create a mark file.
```
Comment (vincent-zurczak) [inline docker-wrapper.sh]: If it does not make sense here, do not keep it. :wink:

File: docker-wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+# Otherwise, every time a container is stopped and restarted, the env.sh file will be updated.
+if [ ! -f docker-mark.txt ]; then
+# put any setup action to be run once here
+	echo "Indicate the Petals configuration was already updated for this container." > docker-mark.txt
+fi
+
+# Launch Petals-cockpit
+/opt/petals-cockpit/petals-cockpit.sh
```
Comment (vincent-zurczak) [inline docker-wrapper.sh]: A very minor thing: please, add an empty at the end of the file (Github generally does not like it).

**Review by vincent-zurczak — CHANGES_REQUESTED** [review]
Good work.
Very few things to modify, mainly the 3 RUN instructions that should be merged together.

File: Dockerfile (hunk 2 above)
Comment (vincent-zurczak) [reply Dockerfile]: Obviously, you should verify what I wrote works.
But something similar must be feasible.

File: Dockerfile (hunk 1 above)
Comment (psouquet) [reply Dockerfile:26]: I changed to a single RUN

File: Dockerfile (hunk 2 above)
Comment (psouquet) [reply Dockerfile]: I changed it like you asked.

**Review by vincent-zurczak — APPROVED** [review]

**Pull request merged**